core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -219,17 +228,20 @@ public StructuredRecord merge(StructuredRecord joinKey, Iterable<JoinElement<Str
     return outRecordBuilder.build();
   }
 
-  void init(Map<String, Schema> inputSchemas) {
-    validateJoinKeySchemas(inputSchemas, conf.getPerStageJoinKeys());
+  void init(Map<String, Schema> inputSchemas, @Nonnull FailureCollector collector) {
```

Comment:
The annotation can be removed. By convention, everything is non-null unless annotated as `@Nullable`.

Reply:
Fixed